tests/fip/test_extractor.py

```diff
@@ -0,0 +1,303 @@
+"""Tests for fiber photometry contract-based extractor."""
```

Comment:
I think in general these tests have way too many mocks and it is not clear what they are testing at all. Most of them are just testing accessing properties (?). Wouldn't it make more sense simply testing the functions that map the data? 
I would also suggest, instead of mocking streams either store a very small chunk of data in the the repo (eg 10 rows of csv). Or even just create the dataframe as a fixture.